### PR TITLE
React map layer admin

### PR DESCRIPTION
**Pull request by jheijari**
### Fixed PR https://github.com/oskariorg/oskari-frontend/pull/829 issues:
* Fixed styling problems
* Fixed storybook example
* Fixed form fields not updating on layer change
* Fixed layer group handling

Refactored state handling by making `AdminLayerForm` functional component.
Moved `AdminLayerFormService` to the flyout.

Removed `DOMPurify` from libs. Using node module instead. (For storybook fix)

**TODO**
#### Handling wfs styles.
Fields not implemented
* StyleJSON
* HoverJSON

#### User interaction handling
* Disabling save & delete buttons while ajax calls are pending.

#### Layer zoom level restrictions
* Vertical slider values doesn't match to the zoom bar

### CODE REVIEW

File: bundles/admin/admin-layereditor/view/AdminLayerFormService.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        }
+
+        me.layer = {
+            version: layer ? layer.getVersion() : '',
+            layer_id: layer ? layer.getId() : null,
+            layerUrl: layer ? layer.getAdmin().url : '',
+            username: layer ? layer.getAdmin().username : '',
+            password: layer ? layer.getAdmin().password : '',
+            layerName: layer ? layer.getLayerName() : '',
+            name_fi: layer ? layer.getName('fi') : '',
```
Comment (ZakarFin) [inline bundles/admin/admin-layereditor/view/AdminLayerFormService.js]: This won't work as we have other locales as well but let's worry about it later.

File: bundles/admin/admin-layereditor/view/AdminLayerFormService.js [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        // Modify layer for backend
+        const layer = { ...this.getLayer() };
+        layer.maplayerGroups = layer.maplayerGroups.map(cur => cur.id).join(',');
+
+        fetch(Oskari.urls.getRoute('SaveLayer'), {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded',
+                'X-XSRF-TOKEN': this.getCookie('XSRF-TOKEN')
```
Comment (ZakarFin) [inline bundles/admin/admin-layereditor/view/AdminLayerFormService.js]: Oskari.app.getXSRFToken() instead.

File: bundles/admin/admin-layereditor/view/AdminLayerFormService.js (hunk 1 above)
Comment (jheijari) [reply bundles/admin/admin-layereditor/view/AdminLayerFormService.js]: Added TODO comments so we won't forget this issue.

**Comment by ZakarFin** [comment]
Work-in-progress but lets merge this before it gets even larger.

File: bundles/admin/admin-layereditor/view/AdminLayerFormService.js (hunk 2 above)
Comment (jheijari) [reply bundles/admin/admin-layereditor/view/AdminLayerFormService.js]: Removed outdated cookie handling.

**Pull request merged**